curated_transformers/generation/falcon.py

```diff
@@ -13,6 +15,18 @@ class FalconGenerator(DefaultGenerator, FromHFHub):
     Generator for Falcon model variants.
     """
 
+    def __init__(self, tokenizer: Tokenizer, causal_lm: FalconCausalLM):
```

Comment:
This is nice for the documentation, but I think we should keep the classes minimal, so no constructor if it's effectively the same as the superclass constructor. If the generated documentation is not clear, we should change the documentation of the superclass constructor.

I don't think people will use the constructor in practice, but instead load it up with a static factory method.

Reply:
>  If the generated documentation is not clear, we should change the documentation of the superclass constructor.

Do you have a suggestion? If we don't override the ctor here, the docstring of the `GeneratorWrapper` ctor will be displayed, but that results in an inconsistency b'ween the Falcon and Dolly v2 generators - The latter doesn't allow a default generation config to be passed to the ctor but the former will. Same with the Llama generator.

Reply:
Good point, let's stick with this.